The gapis server has exited with an error code of: 2

GAPID Version: 1.1.0:76966beae5023f52e2f0dd3878c34a24d0428054
OS: linux amd64

This is on Ubuntu 17.10 with Intel mesa vulkan driver.

Launched my vulkan app, let it draw a few frames, then closed it.
Then I hit Stop Tracing in gapid.
It started loading the trace then this error occurred:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: Panic at command 176688:vkDestroyImage(device: 94441465962608, image: 94441454593360, pAllocator: 0):
runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: 
--- *resolve.FramebufferChangesResolvable ---
 capture:<id:<data:"\034\337n\365b\261\302S\276_\0263\307\221\270O2\235\005\022" > > 
 Store():
  gapis/database/database.go:50  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1

 Build() #0:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1
 Build() #1:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1
 Build() #2:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1
 Build() #3:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1
 Build() #4:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1
 Build() #5:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1

--- *resolve.FramebufferAttachmentResolvable ---
 replaySettings:<device:<id:<data:"<@Y\272)@\275\235M\324PXw+\274-\264!\363\233" > > > after:<capture:<id:<data:"\034\337n\365b\261\302S\276_\0263\307\221\270O2\235\005\022" > > indices:850 > attachment:Color0 settings:<max_width:192 max_height:192 > hints:<preview:true > 
 Store():
  gapis/resolve/framebuffer_attachment.go:59  github.com/google/gapid/gapis/resolve.FramebufferAttachment
  gapis/resolve/thumbnail.go:45  github.com/google/gapid/gapis/resolve.CommandThumbnail
  gapis/resolve/thumbnail.go:33  github.com/google/gapid/gapis/resolve.Thumbnail
  gapis/resolve/resolve.go:324  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:249  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:35  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1

 Build() #0:
  gapis/database/database.go:44  github.com/google/gapid/gapis/database.Resolve
  gapis/resolve/resolve.go:55  github.com/google/gapid/gapis/resolve.ImageInfo
  gapis/resolve/resolve.go:296  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/as.go:29  github.com/google/gapid/gapis/resolve.As
  gapis/resolve/resolve.go:264  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:249  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:35  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1

--- *resolve.GetResolvable ---
 path:<as:<image_format:<uncompressed:<format:<components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:R > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:G > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:B > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:A > > > > image_info:<id:<data:"*;b2\337\371\344\207\236\343\\^\351\324s\030U\004\365\325" > > > > 
 Store():
  gapis/database/database.go:50  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
  gapis/resolve/thumbnail.go:67  github.com/google/gapid/gapis/resolve.CommandThumbnail
  gapis/resolve/thumbnail.go:33  github.com/google/gapid/gapis/resolve.Thumbnail
  gapis/resolve/resolve.go:324  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:249  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:35  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1

 Build() #0:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
  gapis/resolve/thumbnail.go:67  github.com/google/gapid/gapis/resolve.CommandThumbnail
  gapis/resolve/thumbnail.go:33  github.com/google/gapid/gapis/resolve.Thumbnail
  gapis/resolve/resolve.go:324  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:249  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:35  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1

--- *resolve.GetResolvable ---
 path:<thumbnail:<desired_max_width:192 desired_max_height:192 desired_format:<uncompressed:<format:<components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:R > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:G > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:B > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:A > > > > command:<capture:<id:<data:"\034\337n\365b\261\302S\276_\0263\307\221\270O2\235\005\022" > > indices:850 > > > 
 Store():
  gapis/database/database.go:50  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
  gapis/server/server.go:259  github.com/google/gapid/gapis/server.(*server).Get
  gapis/server/grpc.go:167  github.com/google/gapid/gapis/server.(*grpcServer).Get
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto~/github.com/google/gapid/gapis/service/service.pb.go:5656  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
  core/app/auth/auth.go:110  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto~/github.com/google/gapid/gapis/service/service.pb.go:5658  github.com/google/gapid/gapis/service._Gapid_Get_Handler
  external/org_golang_google_grpc/server.go:719  google.golang.org/grpc.(*Server).processUnaryRPC
  external/org_golang_google_grpc/server.go:905  google.golang.org/grpc.(*Server).handleStream

 Build() #0:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
  gapis/server/server.go:259  github.com/google/gapid/gapis/server.(*server).Get
  gapis/server/grpc.go:167  github.com/google/gapid/gapis/server.(*grpcServer).Get
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto~/github.com/google/gapid/gapis/service/service.pb.go:5656  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
  core/app/auth/auth.go:110  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto~/github.com/google/gapid/gapis/service/service.pb.go:5658  github.com/google/gapid/gapis/service._Gapid_Get_Handler
  external/org_golang_google_grpc/server.go:719  google.golang.org/grpc.(*Server).processUnaryRPC
  external/org_golang_google_grpc/server.go:905  google.golang.org/grpc.(*Server).handleStream
 [recovered]
	panic: 
--- *resolve.FramebufferChangesResolvable ---
 capture:<id:<data:"\034\337n\365b\261\302S\276_\0263\307\221\270O2\235\005\022" > > 
 Store():
  gapis/database/database.go:50  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1

 Build() #0:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1
 Build() #1:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1
 Build() #2:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1
 Build() #3:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1
 Build() #4:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1
 Build() #5:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/framebuffer_changes.go:34  github.com/google/gapid/gapis/resolve.FramebufferChanges
  gapis/resolve/framebuffer_attachment.go:75  github.com/google/gapid/gapis/resolve.(*FramebufferAttachmentResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1

--- *resolve.FramebufferAttachmentResolvable ---
 replaySettings:<device:<id:<data:"<@Y\272)@\275\235M\324PXw+\274-\264!\363\233" > > > after:<capture:<id:<data:"\034\337n\365b\261\302S\276_\0263\307\221\270O2\235\005\022" > > indices:850 > attachment:Color0 settings:<max_width:192 max_height:192 > hints:<preview:true > 
 Store():
  gapis/resolve/framebuffer_attachment.go:59  github.com/google/gapid/gapis/resolve.FramebufferAttachment
  gapis/resolve/thumbnail.go:45  github.com/google/gapid/gapis/resolve.CommandThumbnail
  gapis/resolve/thumbnail.go:33  github.com/google/gapid/gapis/resolve.Thumbnail
  gapis/resolve/resolve.go:324  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:249  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:35  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1

 Build() #0:
  gapis/database/database.go:44  github.com/google/gapid/gapis/database.Resolve
  gapis/resolve/resolve.go:55  github.com/google/gapid/gapis/resolve.ImageInfo
  gapis/resolve/resolve.go:296  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/as.go:29  github.com/google/gapid/gapis/resolve.As
  gapis/resolve/resolve.go:264  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:249  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:35  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1

--- *resolve.GetResolvable ---
 path:<as:<image_format:<uncompressed:<format:<components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:R > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:G > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:B > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:A > > > > image_info:<id:<data:"*;b2\337\371\344\207\236\343\\^\351\324s\030U\004\365\325" > > > > 
 Store():
  gapis/database/database.go:50  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
  gapis/resolve/thumbnail.go:67  github.com/google/gapid/gapis/resolve.CommandThumbnail
  gapis/resolve/thumbnail.go:33  github.com/google/gapid/gapis/resolve.Thumbnail
  gapis/resolve/resolve.go:324  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:249  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:35  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1

 Build() #0:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
  gapis/resolve/thumbnail.go:67  github.com/google/gapid/gapis/resolve.CommandThumbnail
  gapis/resolve/thumbnail.go:33  github.com/google/gapid/gapis/resolve.Thumbnail
  gapis/resolve/resolve.go:324  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:249  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:35  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:127  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:214  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1

--- *resolve.GetResolvable ---
 path:<thumbnail:<desired_max_width:192 desired_max_height:192 desired_format:<uncompressed:<format:<components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:R > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:G > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:B > components:<data_type:<integer:<bits:8 > > sampling:<normalized:true > channel:A > > > > command:<capture:<id:<data:"\034\337n\365b\261\302S\276_\0263\307\221\270O2\235\005\022" > > indices:850 > > > 
 Store():
  gapis/database/database.go:50  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
  gapis/server/server.go:259  github.com/google/gapid/gapis/server.(*server).Get
  gapis/server/grpc.go:167  github.com/google/gapid/gapis/server.(*grpcServer).Get
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto~/github.com/google/gapid/gapis/service/service.pb.go:5656  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
  core/app/auth/auth.go:110  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto~/github.com/google/gapid/gapis/service/service.pb.go:5658  github.com/google/gapid/gapis/service._Gapid_Get_Handler
  external/org_golang_google_grpc/server.go:719  google.golang.org/grpc.(*Server).processUnaryRPC
  external/org_golang_google_grpc/server.go:905  google.golang.org/grpc.(*Server).handleStream

 Build() #0:
  gapis/database/database.go:54  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
  gapis/server/server.go:259  github.com/google/gapid/gapis/server.(*server).Get
  gapis/server/grpc.go:167  github.com/google/gapid/gapis/server.(*grpcServer).Get
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto~/github.com/google/gapid/gapis/service/service.pb.go:5656  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
  core/app/auth/auth.go:110  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto~/github.com/google/gapid/gapis/service/service.pb.go:5658  github.com/google/gapid/gapis/service._Gapid_Get_Handler
  external/org_golang_google_grpc/server.go:719  google.golang.org/grpc.(*Server).processUnaryRPC
  external/org_golang_google_grpc/server.go:905  google.golang.org/grpc.(*Server).handleStream

[signal SIGSEGV: segmentation violation code=0x1 addr=0x4c pc=0x1910272]
```

--- a/gapis/api/vulkan/vulkan.api
+++ b/gapis/api/vulkan/vulkan.api
@@ -2993,8 +2993,10 @@ cmd void vkDestroyDevice(
         LastBoundQueue = null
       }
       for i in (0 .. LastPresentInfo.PresentImageCount) {
-        if (LastPresentInfo.PresentImages[i].Device == device) {
-          delete(LastPresentInfo.PresentImages, i)
+        if (LastPresentInfo.PresentImages[i] != null) {
+          if (LastPresentInfo.PresentImages[i].Device == device) {
+            LastPresentInfo.PresentImages[i] = null
+          }
         }
       }
       delete(Queues, v.VulkanHandle)
@@ -4193,8 +4195,10 @@ cmd void vkDestroyImage(
       }
     }
     for i in (0 .. LastPresentInfo.PresentImageCount) {
-      if (LastPresentInfo.PresentImages[i].VulkanHandle == image) {
-        delete(LastPresentInfo.PresentImages, i)
+      if (LastPresentInfo.PresentImages[i] != null) {
+        if (LastPresentInfo.PresentImages[i].VulkanHandle == image) {
+          LastPresentInfo.PresentImages[i] = null
+        }
       }
     }
   }
@@ -7667,7 +7671,9 @@ sub void dovkCmdBeginRenderPass(ref!vkCmdBeginRenderPassArgs args) {
   lastDrawInfo().RenderPass = RenderPasses[args.RenderPass]
   lastDrawInfo().InRenderPass = true
   for _ , _ , v in lastDrawInfo().Framebuffer.ImageAttachments {
-    v.Image.LastBoundQueue = LastBoundQueue
+    if (v.Image != null) {
+      v.Image.LastBoundQueue = LastBoundQueue
+    }
   }
   subpass := lastDrawInfo().RenderPass.SubpassDescriptions[0]
   for _ , _ , a in subpass.InputAttachments {

